### DISCUSSION TITLE
How to use emulator not more than once in the development?

### DISCUSSION THREAD

**Discussion by raviSussol** [invertase/react-native-firebase#6276] [Q&A]
I have been using the emulator to work with the firebase locally. I have setup the emulator configuration like below:
```
const connectToEmulators = () => {
    firestore().useEmulator('localhost', 8080);
    auth().useEmulator('http://localhost:9099');
    functions().useEmulator('localhost', 5001);
    storage().useEmulator('localhost', 9199);
};

if (__DEV__) {
    connectToEmulators();
  }
```

Now when I run the app the warning message appears below:
`Cannot call useEmulator() after instance has already been initialized.`

The message itself clear that the warning is due the emulators are being used multiple time due to react re-render behavior. But how can I make the emulator run only once to avoid this warning message in the startup? Any help would be appreciated.

**Comment by mikehardy** [comment]
#5723 / #5860 dupe - no way currently, needs a PR to either store emulator usage state at native level in order to correctly ignore the error or just skip the underlying native call if useEmulator was already called and it's being called again

**Reply by raviSussol** [reply]
Thanks @mikehardy for the info. I've been using a global boolean variable to flag the emulator started state but still getting this warning message when doing refresh cause the global variable also gets reset after the screen is refreshed so had an idea of keeping emulator state in somewhere like async storage to persist state even if the screen is refreshed. But before doing that was curious to know as if there is already an alternative option that this library offers. No worries.

**Reply by mikehardy** [reply]
The global variable has to be at the native level, not JavaScript
